nx-plugin/src/generators/shared/generator.utils.ts

```
@@ -1,24 +1,73 @@
 import { Tree } from '@nx/devkit';
 import ora = require('ora');
 
+export interface GeneratorStepError {
+  error: string;
+  stopExecution: boolean;
+}
+
 export interface GeneratorStep<T> {
-  process(tree: Tree, options: T): void;
+  process(tree: Tree, options: T): void | GeneratorStepError;
```

I am not sure if returning void or object is a good idea, seems a little bit odd. Maybe GeneratorStepSuccess object would be sufficient or something?

---

Hmm, I mean theoretically we could also throw an error inside and catch it at the level above, maybe that makes more sense?

---

Seems better imo